### PR TITLE
Remove variables and logic referencing artifactory (caching proxy)

### PR DESCRIPTION
**Pull request by alopezz**
### What does this PR do?

Removes all references to the caching mirror that we've stopped using.

### Motivation

[BARX-420](https://datadoghq.atlassian.net/browse/BARX-420J)

### Describe how to test/QA your changes

[BARX-420]: https://datadoghq.atlassian.net/browse/BARX-420?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -375,33 +369,11 @@ variables:
 # Workflow rules
 # Rules used to define whether a pipeline should run, and with which variables
 #
-
-# WARNING: Do not change below if you want to globally disable the caching proxy (all branches would need to be rebased for it to be effectively disabled).
-# Instead use the break-glass mechanism by looking for the "Remotely disable Artifactory" doc in Confluence.
```
Comment (spencergilbert) [inline .gitlab-ci.yml:380]: **❓ question:** ‏ do we still need/have a break-glass mechanism to disable using the new caching solution?

**Review by spencergilbert — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (alopezz) [reply .gitlab-ci.yml:380]: I don't think anything like that would be applicable to the pieces that are partially replacing the role of the caching mirror.

File: .gitlab/common/shared.yml [hunk 2]
```diff
@@ -13,28 +13,6 @@
   - $ARTIFACTORY_BYPASS=$(& "$env:CI_PROJECT_DIR\tools\ci\aws_ssm_get_wrapper.ps1" $ARTIFACTORY_BYPASS_SSM_NAME)
   - if ($Env:ARTIFACTORY_BYPASS -eq "true") { Write-Host "Bypassing Artifactory" }
```
Comment (amenasria) [inline .gitlab/common/shared.yml:15]: **💬 suggestion:** ‏I believe we should remove the code above as well

**Review by amenasria — CHANGES_REQUESTED** [review]
LGTM but I think we could also remove the artifactory bypass logic ! WDYT?

**Review by amenasria — APPROVED** [review]

**Comment by CelianR** [comment]
## Gitlab CI Configuration Changes

File: .gitlab/common/shared.yml (hunk 2 above)
Comment (alopezz) [reply .gitlab/common/shared.yml:15]: Thanks for the catch! I somehow missed this reference while grepping.

**Review by amenasria — APPROVED** [review]
LGTM !!


**👏 praise:** ‏Thanks for this !!!

**Review by FlorentClarret — APPROVED** [review]
Nice!

**Pull request merged**